quinn-proto/src/connection/stats.rs

```diff
@@ -0,0 +1,80 @@
+/// Statistics about packets transmitted on a connection
+#[derive(Default)]
+pub struct PacketTransmissionStats {
+    /// The amount of packets transmitted on a connection
+    pub packets: u64,
+    /// The total amount of bytes transmitted on a connection
+    pub bytes: u64,
+}
+
+impl PacketTransmissionStats {
+    /// Calculates the average packet size
+    pub fn avg_packet_size(&self) -> u64 {
+        if self.packets != 0 {
+            self.bytes / self.packets
+        } else {
+            0
+        }
+    }
+}
+
+impl std::fmt::Debug for PacketTransmissionStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PacketTransmissionStats")
+            .field("number of packets", &self.packets)
+            .field("number of bytes", &self.bytes)
+            .field("avg packet size", &self.avg_packet_size())
+            .finish()
+    }
+}
+
+/// Statistics about frames transmitted on a connection
+#[derive(Default)]
+pub struct FrameTransmissionStats {
+    pub acks: u64,
```

Comment:
We can likely shrink down most of those to `u32` without danger. Or use `u32` with saturating increments

Reply:
I'd favor sticking with `u64` and just not worrying about it.